aws-datastore/src/test/java/com/amplifyframework/datastore/AWSDataStorePluginTest.java

```diff
@@ -158,29 +168,117 @@ public void configureAndInitializeInApiModeWithoutApi() throws JSONException, Am
+            awsDataStorePlugin.save(person1, itemSaved -> {
+                assertNotNull(itemSaved.item().getId());
+                assertEquals(person1.getLastName(), itemSaved.item().getLastName());
+                single.onSuccess(true);
+            }, single::onError);
+        }).andThen(
+            Completable.fromSingle(single -> { // Verify the record has been saved
+                awsDataStorePlugin.query(Person.class, results -> {
+                    Person actualPerson = results.next();
+                    assertNotNull(actualPerson);
+                    assertFalse(results.hasNext()); // We should only have one result.
+                    assertEquals(person1, actualPerson);
+                    single.onSuccess(true);
+                }, single::onError);
+            })
+        ).andThen(
+            Completable.fromSingle(single -> { // Clear the local store.
+                awsDataStorePlugin.clear(() -> {
+                    // Make sure the remote subscription operations were cancelled
+                    assertRemoteSubscriptionsCancelled();
+                    single.onSuccess(true);
+                }, single::onError);
+            })
+        ).andThen(
+            Completable.fromSingle(single -> { // Save a new record to local store
+                awsDataStorePlugin.save(person2, itemSaved -> {
+                    assertEquals(person2.getLastName(), itemSaved.item().getLastName());
+                    // Check if the sync process restarted.
+                    assertRemoteSubscriptionsStarted();
+                    single.onSuccess(true);
+                }, single::onError);
+            })
+        ).andThen(
+            Completable.fromSingle(single -> { // Verify the record has been saved
+                awsDataStorePlugin.query(Person.class, results -> {
+                    Person actualPerson = results.next();
+                    assertNotNull(actualPerson);
+                    assertFalse(results.hasNext()); //we should only have one result.
+                    assertEquals(person2, actualPerson);
+                    single.onSuccess(true);
+                }, single::onError);
+            })
+        ).blockingGet(OPERATION_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+
+        // Verify that the API mutate method was called once for each model saved.
+        verify(mockApiCategory, times(2)).mutate(Mockito.any(), Mockito.any(), Mockito.any());
+    }
+
+    private void assertRemoteSubscriptionsCancelled() {
+        // Check that we've had active subscriptions
+        assertTrue(subscriptionStartedCounter.get() > 0);
+        // And the number of started and cancelled are the same
+        assertEquals(subscriptionStartedCounter.get(), subscriptionCancelledCounter.get());
+    }
 
-        assertTrue(syncProcessorInvoked);
+    private void assertRemoteSubscriptionsStarted() {
+        // For each model, there should be 3 subscriptions setup.
```

Comment:
You could phrase this as,
```
final int diffTypesOfSubscriptions  = SubscriptionType.values().length;
```
and then get out of hte business of writing `3`, while also clarifying its meaning.